cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -393,7 +402,14 @@ public void getPluginsInfo(HttpRequest request, HttpResponder responder,
     throws NamespaceNotFoundException, BadRequestException, ApplicationNotFoundException, AccessException {
 
     ApplicationId applicationId = validateApplicationId(namespaceId, appId);
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(applicationLifecycleService.getPlugins(applicationId)));
+    try {
+      String latestAppVersion = getLatestAppVersion(applicationId);
+      applicationId = validateApplicationVersionId(namespaceId, appId, latestAppVersion);
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(applicationLifecycleService.getPlugins(applicationId)));
+    } catch (Exception e) {
+      LOG.error("Failure to retrieve plugin info", e);
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, e.getMessage());
```

Comment:
Why are you mapping it to BAD_REQUEST? Generally don't do custom error code serving unless absolutely needed. It should be done by a general error handler in a consistent manner based on exception data

Reply:
I see, okay so I'll remove the try-catch block